### DISCUSSION TITLE
Dataset for index page

### DISCUSSION THREAD

**Discussion by im-redactd** [aminalaee/sqladmin#750] [Q&A] [answered]
Is there a mechanism to provide the index.html with a dataset? I'd like to be able to have some metrics shown on the initial landing page if possible.

**Comment by im-redactd** [comment]
I suppose I'm making things harder than I need to. Initially I was trying to `@expose("/")` somehow to give the initial page extra context but I think the answer to my question is to just create something like /admin/home and be done with it.

**Comment by aminalaee** [comment, marked as answer]
https://github.com/aminalaee/sqladmin/blob/main/sqladmin/templates/sqladmin/index.html
You can override this template to achieve that.

**Reply by im-redactd** [reply]
I'm doing that now by using a custom view like this. 
```
@expose("/home", methods=["GET"])
    async def test_page(self, request: Request):
        custom_data = {
            "message": "Welcome to the Admin Dashboard!",
            "stats": {"users": 10, "posts": 50}
        }
        return await self.templates.TemplateResponse(request,"index.html", 
            context=custom_data
        )
```

Is it possible to provide custom data to the base url /admin/ ? or do I need to continue to do /admin/home to achieve what I need?

thank you

**Reply by aminalaee** [reply]
Yes, that's what I meant. Instead of creating a new view, you want to override "index.html" because that's the empty page.
